lxd/nsexec.go

```diff
@@ -162,6 +162,18 @@ int manip_file_in_ns(char *rootfs, int pid, char *host, char *container, bool is
 	if (is_put && stat(container, &st) < 0)
 		exists = 0;
 
+	if (is_put) {
+		char *pos = strrchr(container, '/');
+		if (pos) {
+			*pos = 0;
+			if (mkdir_p(container, 0755) < 0) {
```

Comment:
No, mode here is the mode of the actual file to be created. We need +x in the directory to be able to manipulate its children, so if the user specified mode is e.g. 600, subsequent mkdirs will fail. I suppose we could mask on +x to whatever the user specified mode is if we really wanted to.